internal/elasticsearch/common_test.go

```diff
@@ -564,6 +565,47 @@ func TestPodSpecHasTaintTolerations(t *testing.T) {
 	}
 }
 
+func TestElasticSearchSecurityContext(t *testing.T) {
+	podTemplate := newPodTemplateSpec(context.Background(), log.NewLogger("common-testing"), "test-node-name", "test-cluster-name", "test-namespace-name", api.ElasticsearchNode{}, api.ElasticsearchNodeSpec{}, map[string]string{}, map[api.ElasticsearchNodeRole]bool{}, nil, LogConfig{})
+
+	expectedPod := v1.PodSecurityContext{
+		RunAsNonRoot: pointer.Bool(true),
+	}
+
+	if diff := cmp.Diff(*podTemplate.Spec.SecurityContext, expectedPod); diff != "" {
```

Comment:
I'd probably define the `expectedPod` variable as a pointer as well, so that I could pass them both as pointers to the Diff. A `nil` SecurityContext would cause a panic now.